src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -136,15 +137,21 @@ public static int OffsetToStringData
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern bool TryEnsureSufficientExecutionStack();
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object GetUninitializedObjectInternal(
-            // This API doesn't call any constructors, but the type needs to be seen as constructed.
-            // A type is seen as constructed if a constructor is kept.
-            // This obviously won't cover a type with no constructor. Reference types with no
-            // constructor are an academic problem. Valuetypes with no constructors are a problem,
-            // but IL Linker currently treats them as always implicitly boxed.
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
-            Type type);
+        private static unsafe object GetUninitializedObjectInternal(Type type)
```

Comment:
__Note__: Had to remove `[DynamicallyAccessedMembers]` because it was causing the mono linker to crash. (Perhaps due to the use of _calli_ in this method?)

The attribute is still on the public entry point to this method, so it should still be ok?

Reply:
Could you please report that as issue at mono/linker

Reply:
Can you attach a CoreLib that crashes linker here? I'll take a look.

The reason why the internal API is annotated is to ensure that if someone adds a new callsite to this API, they get a warning if things are not properly annotated. We want to prevent a situation where things are broken at runtime and there is no warning.

Reply:
Actually, I don't need a CoreLib. I know what this is: 

https://github.com/mono/linker/blob/f1aadabc832eaad6be133a13d5bb69ef2ab0011f/src/linker/Linker.Dataflow/MethodBodyScanner.cs#L509-L511

Reply:
As a workaround can I put the attribute on the `RuntimeType` parameter at the newly introduced internal method on `RuntimeHandles`?
https://github.com/dotnet/runtime/blob/e09e91bbc4581d33b07ba70f8d808b2e1d8bc8ea/src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs#L217

Reply:
Yep, we should annotate that because it's an API that lets one create new types without linker otherwise seeing that (linker it going to e.g. stub out virtual methods on types that are not seen as allocated).

Annotating that one and removing annotation here will trigger a warning here. At least we won't forget about restoring the annotation here.

Reply:
Thanks Michal for tracking this! :)
I'll wait for things to flow through.